### PR TITLE
process-agent,system-probe: Add an internal_deploy stage to the pipeline

### PR DESCRIPTION
**Pull request by sfluor**
### What does this PR do?

Add a manual step to deploy the process-agent and system-probe binaries internally

### Motivation

Faster and easier testing / development cycles.

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1185,6 +1187,20 @@ deploy_puppy:
     - export PACKAGE_VERSION=$(inv agent.version --url-safe)
     - aws s3 cp --region us-east-1 ./agent $S3_DSD6_URI/puppy/agent-$PACKAGE_VERSION --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers full=id=3a6e02b08553fd157ae3fb918945dd1eaae5a1aa818940381ef07a430cf25732
 
+# deploy process agent and system-probe to staging bucket
+deploy_process_and_sysprobe:
+  stage: internal_deploy
+  when: manual
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/deploy:latest
+  before_script:
+    - cd $OMNIBUS_PACKAGE_DIR
```
Comment (leeavital) [inline .gitlab-ci.yml]: Does the cd propagate to the next comane? Elsewhere is: `    - ls $OMNIBUS_PACKAGE_DIR` (all one command)

https://github.com/DataDog/datadog-agent/pull/3559/files#diff-96edf7a6f008de9e928d04e1ae5e12a5R1183

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1185,6 +1187,20 @@ deploy_puppy:
...
+    - ls
+  tags: [ "runner:main", "size:large" ]
+  script:
+    - dpkg -x datadog-agent_*_amd64.deb ./out
+    - $S3_CP_CMD ./out/opt/datadog-agent/embedded/bin/process-agent s3://$PROCESS_S3_BUCKET/process-agent-amd64-${CI_COMMIT_SHA:0:7}
```
Comment (leeavital) [inline .gitlab-ci.yml]: Thoughts on also including the ref name here?

File: .gitlab-ci.yml (hunk 1 above)
Comment (sfluor) [reply .gitlab-ci.yml]: It does for the rest of the script 👍

**Review by leeavital — APPROVED** [review]

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -1185,6 +1187,24 @@ deploy_puppy:
     - export PACKAGE_VERSION=$(inv agent.version --url-safe)
     - aws s3 cp --region us-east-1 ./agent $S3_DSD6_URI/puppy/agent-$PACKAGE_VERSION --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers full=id=3a6e02b08553fd157ae3fb918945dd1eaae5a1aa818940381ef07a430cf25732
 
+# deploy process agent and system-probe to staging bucket
+deploy_process_and_sysprobe:
```
Comment (sunhay) [inline .gitlab-ci.yml]: `s/deploy/upload`?

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -1185,6 +1187,24 @@ deploy_puppy:
...
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/deploy:latest
+  before_script:
+    - cd $OMNIBUS_PACKAGE_DIR
+    - ls
+  tags: [ "runner:main", "size:large" ]
+  script:
+    - dpkg -x datadog-agent_*_amd64.deb ./out
+    # Use tag or shortened branch with short commit hash to identify the binary
+    - export SHORT_REF=$(echo $CI_COMMIT_REF_NAME | cut -d'/' -f2- | cut -c -10 | sed -E 's/[^[:alnum:]]+/-/g')
+    - export NAME="${CI_COMMIT_TAG:-$SHORT_REF}-${CI_COMMIT_SHA:0:7}"
```
Comment (sunhay) [inline .gitlab-ci.yml]: Do you have any examples of what these names will look like?

Comment (sfluor) [reply .gitlab-ci.yml]: ```process-agent-amd64-process-ag-5c6a4e2```

File: .gitlab-ci.yml (hunk 3 above)
Comment (sfluor) [reply .gitlab-ci.yml]: The other steps are using `deploy` I think it's better to keep it consistent

File: .gitlab-ci.yml (hunk 4 above)
Comment (sunhay) [reply .gitlab-ci.yml]: This is a messy name. Any reason we can't just use `process-agent-amd64-5c6a4e2`

Comment (sunhay) [reply .gitlab-ci.yml]: Looks like @leeavital reccommended adding the ref. What use is it?

**Pull request merged**